### PR TITLE
Feat: minor feed improvement

### PR DESCRIPTION
**Pull request by MrPunyapal**

### CODE REVIEW

File: app/Queries/Feeds/QuestionsFollowingFeed.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ public function builder(): Builder
     {
         $followQueryClosure = function (Builder $query): void {
             $query->where('to_id', $this->user->id)
-                ->orWhereIn('to_id', $this->user->following()->select('users.id'));
+                ->orWhereIn('to_id', DB::table('followers')->select('user_id')->where('follower_id', $this->user->id));
```
Comment (MrPunyapal) [inline app/Queries/Feeds/QuestionsFollowingFeed.php:30]: this helps to remove unwanted inner join due to `belongsToMany` relation.

File: resources/views/components/thread.blade.php [hunk 2]
```diff
@@ -13,30 +13,30 @@
             :in-thread="true"
             :key="'question-'.$rootId"
         />
+
+        @if($grandParentId !== null && ($parentId === null || $grandParentId !== $rootId))
```
Comment (MrPunyapal) [inline resources/views/components/thread.blade.php:17]: here we are adding support for following (when root is from following user and parent is not from following user it adds view more comments divider in thread.)

File: app/Queries/Feeds/QuestionsFollowingFeed.php (hunk 1 above)
Comment (msamgan) [reply app/Queries/Feeds/QuestionsFollowingFeed.php:30]: I'll suggest ``whereIntegerInRaw`` in please if WhereIn.

image

File: resources/views/components/thread.blade.php (hunk 2 above)
Comment (nunomaduro) [reply resources/views/components/thread.blade.php:17]: What `grandParentId` means?

**Review by nunomaduro — CHANGES_REQUESTED** [review]

File: resources/views/components/thread.blade.php (hunk 2 above)
Comment (MrPunyapal) [reply resources/views/components/thread.blade.php:17]: > What `grandParentId` means?

The question's root and the question's parent's patent (means grand parent) both are the same we do not show 'view more comments' as there are no comments between both questions.

File: app/Queries/Feeds/QuestionsFollowingFeed.php (hunk 1 above)
Comment (MrPunyapal) [reply app/Queries/Feeds/QuestionsFollowingFeed.php:30]: There are no bindings 🤞

File: resources/views/components/thread.blade.php (hunk 2 above)
Comment (MrPunyapal) [reply resources/views/components/thread.blade.php:17]: Or it should be based on comments count on root? 🤔

**Pull request merged**